mlir/lib/Dialect/SparseTensor/IR/SparseTensorDialect.cpp

```diff
@@ -438,14 +438,15 @@ SparseTensorEncodingAttr::tranlateShape(ArrayRef<int64_t> srcShape,
     // Do constant propagation on the affine map.
     AffineExpr evalExp =
         simplifyAffineExpr(exp.replaceDims(dimRep), srcShape.size(), 0);
-    if (auto c = evalExp.dyn_cast<AffineConstantExpr>()) {
+    // use llvm namespace here to avoid ambiguity
+    if (auto c = llvm::dyn_cast<AffineConstantExpr>(evalExp)) {
```

Comment:
can we avoid the llvm:: prefix if possible? it does not seem to be needed below, so I would like to keep the shorter, consistent version

Reply:
> can we avoid the llvm:: prefix if possible? it does not seem to be needed below, so I would like to keep the shorter, consistent version

Because the compiler reports that it cannot find this function, I will check where the namespace is missing.

Reply:
if we did't use qualified name `llvm::dyn_cast` compiler found the Attribute's dyn_cast function and abort the name found.then report argument not match.I think this can safe remove untile all of the deprecated T.cast style functions removed.
name find rule: https://en.cppreference.com/w/cpp/language/lookup

Reply:
ah, I see. it just looked strange on some, but not all occurrences; but if the compiler tells you to do so, no choice ;-)